dashboard/app/models/school_district.rb

```diff
@@ -36,37 +36,74 @@ def self.get_seed_filename(stub_school_data)
   # @param options [Hash] Optional map of options.
   def self.seed_all(options = {})
     options[:stub_school_data] ||= CDO.stub_school_data
-    options[:force] ||= false
 
-    # use a much smaller dataset in environments that reseed data frequently.
-    school_districts_tsv = get_seed_filename(options[:stub_school_data])
-    expected_count = `wc -l #{school_districts_tsv}`.to_i - 1
-    raise "#{school_districts_tsv} contains no data" unless expected_count > 0
-
-    # It takes approximately 30 seconds to seed config/school_districts.tsv.
-    # Skip seeding if the data is already present. Note that this logic will
-    # not re-seed data if the number of records in the DB is greater than or
-    # equal to that in the TSV file, even if the data is different.
-    # Stubbed data is small enough to seed it every time.
-    if options[:stub_school_data] || options[:force] || SchoolDistrict.count < expected_count
-      CDO.log.debug "seeding school districts (#{expected_count} rows)"
+    if options[:stub_school_data]
+      # use a much smaller dataset in environments that reseed data frequently.
+      school_districts_tsv = get_seed_filename(true)
       SchoolDistrict.transaction do
         merge_from_csv(school_districts_tsv)
       end
+    else
+      SchoolDistrict.seed_from_s3
+    end
+  end
+
+  def self.seed_from_s3
+    SchoolDistrict.transaction do
+      CDO.log.info "Seeding 2013-2014 school district data"
+      AWS::S3.seed_from_file('cdo-nces', "2013-2014/ccd/ag131a_supp.txt") do |filename|
+        SchoolDistrict.merge_from_csv(filename) do |row|
```

Comment:
Since this district text file was already imported using a different method, shouldn't we protect this call to `merge_from_csv` by setting the `write_updates` argument to `false`?  Same with the other data files lower down.